ext/opcache/Optimizer/zend_optimizer.c

```diff
@@ -531,13 +531,6 @@ int zend_optimizer_update_op2_const(zend_op_array *op_array,
 		case ZEND_FETCH_DIM_UNSET:
 		case ZEND_FETCH_LIST_R:
 		case ZEND_FETCH_LIST_W:
-			if (Z_TYPE_P(val) == IS_STRING) {
-				zend_ulong index;
-				if (ZEND_HANDLE_NUMERIC(Z_STR_P(val), index)) {
-					zval_ptr_dtor_nogc(val);
-					ZVAL_LONG(val, index);
-				}
-			}
```

Comment:
This code can be left for ADD_ARRAY_ELEMENT, and the rest can simply be dropped from the switch. They'll fall through to the default case.

Reply:
Done in 6a4ae15587